### PR TITLE
Environment variables management

### PR DESCRIPTION
**Pull request by lovincyrus**
https://www.notion.so/rilldata/Environment-Variables-management-UI-42304f145b07410d86c2a523496aa6bc

https://www.notion.so/rilldata/DEV-NOTE-12eba33c8f57804ba998e1368c391fd6

https://www.figma.com/design/Qt6EyotCBS3V6O31jVhMQ7/RILL-Latest?node-id=17068-794459&node-type=instance&t=LLS6oUiG9I9OqemC-0

https://github.com/user-attachments/assets/be4aae56-a7f1-4f8d-a7ac-70cd9767e540

### CODE REVIEW

File: web-admin/src/features/projects/environment-variables/ValueCell.svelte [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+  }
+
+  function toggleShowValue() {
+    showValue = !showValue;
+  }
+
+  function copyToClipboard(text: string) {
+    navigator.clipboard.writeText(text).catch(console.error);
+  }
```
Comment (ericpgreen2) [inline web-admin/src/features/projects/environment-variables/ValueCell.svelte:35]: What about the `copyToClipboard` function in `web-common/src/lib/actions/copy-to-clipboard.ts`?

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]
Code looks great!

UXQA:
1. Importing a `.env` file results in an extra, blank entry at the top of the list
2. The dialog should overflow & scroll when there are many env variables. Currently the environment variables are hidden offscreen.
    image
3. The filters aren't working for environment variables that belong to both Development and Production environments. E.g., a filter for Development variables should include a variable that belongs to both Dev & Prod environments. [See Jam](https://jam.dev/c/cf3ee023-7f18-4bef-8737-c752b13dd4cd)
4. In the Add Environment Variables form, after filling out n rows, clicking the nth row's  "trash" icon will delete all rows, not just the nth. [See Jam](https://jam.dev/c/07e5db8e-7ad5-44a7-a150-35ec093c09db)

**Comment by lovincyrus** [comment]
> * The filters aren't working for environment variables that belong to both Development and Production environments. E.g., a filter for Development variables should include a variable that belongs to both Dev & Prod environments. [See Jam](https://jam.dev/c/cf3ee023-7f18-4bef-8737-c752b13dd4cd)

I implemented the filter as All environments, production only, and development only.

**Comment by ericokuma** [comment]
Ah @lovincyrus, my apologies. I checked the PRD and it wasn't clear. The expected behavior for filters is as @ericpgreen2 outlined i.e. All environments, includes production, includes development

**Comment by lovincyrus** [comment]
Got it, updated! @ericokuma @ericpgreen2

**Comment by ericokuma** [comment]
thank you!

**Review by ericpgreen2 — COMMENTED** [review]
Can we sort the environment variables alphabetically? I was confused when I edited the environment of a variable (I added the "Production" environment), and the variable moved to a different location in the list. At first I thought my edit wasn't reflected. [Jam](https://jam.dev/c/9a6caf67-f2a5-4015-85df-12e482d9127b).

File: web-admin/src/features/projects/environment-variables/EditDialog.svelte [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    id: id,
+    SPA: true,
+    validators: schema,
+    async onUpdate({ form }) {
+      if (!form.valid) return;
+      const values = form.data;
+
+      const flatVariables = {
+        [values.key]: values.value,
+      };
```
Comment (ericpgreen2) [inline web-admin/src/features/projects/environment-variables/EditDialog.svelte:97]: Nit: should this be the singular `flatVariable`?

File: web-common/src/components/forms/Input.svelte [hunk 3]
```diff
@@ -33,6 +33,7 @@
   export let fields: string[] | undefined = [];
   export let disabled = false;
   export let disabledMessage = "No valid options";
+  export let readonly = false;
```
Comment (ericpgreen2) [inline web-common/src/components/forms/Input.svelte:36]: Are there any `<Input />` instances using this prop? I'm not seeing it used at first glance.

**Review by ericpgreen2 — APPROVED** [review]
Great! Approving for you to merge when ready.

File: web-common/src/components/forms/Input.svelte (hunk 3 above)
Comment (lovincyrus) [reply web-common/src/components/forms/Input.svelte:36]: previously added variable name to be readonly, removing it now

File: web-admin/src/features/projects/environment-variables/EditDialog.svelte (hunk 2 above)
Comment (lovincyrus) [reply web-admin/src/features/projects/environment-variables/EditDialog.svelte:97]: right, singular, updating it

**Pull request merged**